Extension for camel-threadpoolfactory-vertx #1430

JiriOndrusek opened the pull request:
Fixes: https://github.com/apache/camel-quarkus/issues/1430

[ ] An issue should be filed for the change unless this is a trivial change (fixing a typo or similar). One issue should ideally be fixed by not more than one commit and the other way round, each commit should fix just one issue, without pulling in other changes.
[ ] Each commit in the pull request should have a meaningful and properly spelled subject line and body. Copying the title of the associated issue is typically enough. Please include the issue number in the commit message prefixed by #.
[ ] The pull request description should explain what the pull request does, how, and why. If the info is available in the associated issue or some other external document, a link is enough.
[ ] Phrases like Fix #<issueNumber> or Fixes #<issueNumber> will auto-close the named issue upon merging the pull request. Using them is typically a good idea.
[ ] Please run mvn process-resources -Pformat (and amend the changes if necessary) before sending the pull request.
[ ] Contributor guide is your good friend: https://camel.apache.org/camel-quarkus/latest/contributor-guide.html

aldettinger commented on docs/modules/ROOT/pages/extensions/threadpoolfactory-vertx.adoc:
```diff
@@ -0,0 +1,28 @@
+
+== What's inside
+
+* https://camel.apache.org/components/latest/others/threadpoolfactory-vertx.html[ThreadPoolFactory Vert.x]
```
threadpoolfactory-vertx.html: is it https://camel.apache.org/components/latest/others/reactive-threadpoolfactory-vertx.html ?

ppalaga replied:
Filed a followup https://github.com/apache/camel-quarkus/issues/1506

jamesnetherton commented on .github/test-categories.yaml:
```diff
@@ -105,6 +105,7 @@ platform:
   - platform-http-engine
   - reactive-streams
   - rest-openapi
+  - threadpoolfactory-vertx
```
Because there is no dedicated integration test module for `threadpoolfactory-vertx`. You may need to add it to the list of exclusions in `scripts/validate-github-workflows.groovy` in order to prevent the validation checks from failing.

JiriOndrusek commented:
@aldettinger I've fixed doc link according your suggestion (from the content of the suggested page it is the correct page)
@jamesnetherton I've removed this module from `test-categories.yaml` because it doesn't have a separate testing module, is this state correct?

jamesnetherton commented:
> is this state correct

Yep - if the itest module does not exist, then there's no need to reference it in `test-categories.yaml`. Your new tests are in `main` which is already specified in that file.

JiriOndrusek commented:
Test failure is not connected to this change: 
>There are uncommitted changes
HEAD detached at pull/1462/merge
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   docs/modules/ROOT/pages/extensions/aws-s3.adoc
	modified:   docs/modules/ROOT/pages/extensions/aws2-s3.adoc
	modified:   docs/modules/ROOT/pages/list-of-camel-quarkus-extensions.adoc

Restarting test (locally after build from test revision, there is nothing unstaged)

JiriOndrusek commented:
Check problem is caused by failing debezium test. Error is caused by an issue in debezium 1.2.0.Final - https://github.com/apache/camel-quarkus/issues/1490

davsclaus commented:
@JiriOndrusek can you fix the pom.xml conflict

davsclaus commented:
This PR also has a commit to update to 3.5-SNAPSHOT of camel, is this really needed?

JiriOndrusek commented:
@davsclaus I'll fix it. PR is meant for branch camel-master, so there should already be 3.5-SNAPSHOT version of camel in pom. I'll fix the conflict and look why it  looks differently.

JiriOndrusek commented:
Test fails because of https://github.com/apache/camel-quarkus/issues/1493

ppalaga commented on extensions-core/core/runtime/src/main/java/org/apache/camel/quarkus/core/BaseModel.java:
```diff
@@ -110,6 +112,91 @@ public synchronized RouteDefinition getRouteDefinition(String id) {
+
+    @Override
+    public RouteTemplateDefinition getRouteTemplateDefinition(String id) {
+        for (RouteTemplateDefinition route : routeTemplateDefinitions) {
+            if (route.idOrCreate(camelContext.adapt(ExtendedCamelContext.class).getNodeIdFactory()).equals(id)) {
```
```suggestion
        final ... nodeIdFactory = camelContext.adapt(ExtendedCamelContext.class).getNodeIdFactory();
        for (RouteTemplateDefinition route : routeTemplateDefinitions) {
            if (route.idOrCreate(nodeIdFactory).equals(id)) {
```

It would be more effective to get the factory once before starting the loop.

JiriOndrusek replied:
No problem, I'll refactor it

ppalaga commented on extensions-core/core/runtime/src/main/java/org/apache/camel/quarkus/core/BaseModel.java:
```diff
@@ -110,6 +112,91 @@ public synchronized RouteDefinition getRouteDefinition(String id) {
+
+    @Override
+    public void addRouteTemplateDefinition(RouteTemplateDefinition routeTemplateDefinition) throws Exception {
+        this.routeTemplateDefinitions.addAll(routeTemplateDefinitions);
```
Does not look right. Perhaps this?
```suggestion
        this.routeTemplateDefinitions.addAll(routeTemplateDefinition);
```

JiriOndrusek replied:
I agree that current state doesn't make sense. I'll fix it.

ppalaga commented on extensions-core/core/runtime/src/main/java/org/apache/camel/quarkus/core/BaseModel.java:
```diff
@@ -110,6 +112,91 @@ public synchronized RouteDefinition getRouteDefinition(String id) {
+
+    @Override
+    public void removeRouteTemplateDefinitions(Collection<RouteTemplateDefinition> routeTemplateDefinitions) throws Exception {
+        routeTemplateDefinitions.removeAll(routeTemplateDefinitions);
```
Does not look right either. Maybe this?
```suggestion
        this.routeTemplateDefinitions.removeAll(routeTemplateDefinitions);
```

JiriOndrusek commented:
@ppalaga I'll rebase it. I think, that the last 2 change requests will disappear, as they were not the part of the original changes, but it seems that they were added during some  rebases. 
I'll fix the first change request (about performance)

JiriOndrusek commented:
@ppalaga I see that these 2 last issues are already in camel-master, I'll fix both of them.
https://github.com/apache/camel-quarkus/blob/camel-master/extensions-core/core/runtime/src/main/java/org/apache/camel/quarkus/core/BaseModel.java#L140
https://github.com/apache/camel-quarkus/blob/camel-master/extensions-core/core/runtime/src/main/java/org/apache/camel/quarkus/core/BaseModel.java#L145

JiriOndrusek commented:
@ppalaga I've embedded all 3 fixes into this PR (they are all from current camel-master branch) I think that creating a simple PR for all 3 of them could be better solution, what do you think?

ppalaga commented:
> I've embedded all 3 fixes into this PR (they are all from current camel-master branch) I think that creating a simple PR for all 3 of them could be better solution, what do you think?

No problem to have unrelated changes in one PR, as long as they are in separate commits.

JiriOndrusek commented:
@ppalaga I'll change it into separate commit. Now it is amended

JiriOndrusek commented:
Good to merge from my point of view

lburgazzoli commented:
@ppalaga @jamesnetherton ok to merge ?

ppalaga commented:
I will take care
